src/PrinterStatus.php

```diff
@@ -0,0 +1,412 @@
+    const POS_FUNCTION_SETTINGS = 12;
+    const POS_HEAD_UP = 14;
+    const POS_RIBBON_OUT = 15;
+    const POS_THERMAL_TRANSFER_MODE = 16;
+    const POS_PRINT_MODE = 17;
+    const POS_PRINT_WIDTH_MODE = 18;
+    const POS_LABEL_WAITING = 19;
+    const POS_LABELS_REMAINING = 20;
+    const POS_FORMAT_WHILE_PRINTING = 21;
+    const POS_IMAGES_IN_MEMORY = 22;
+    const POS_PASSWORD = 23;
+    const POS_STATIC_RAM_INSTALLED = 24;
+    
+    const BAUD_CODES = [
+        '0000' => 110,
+        '0001' => 300,
+        '0010' => 600,
+        '0011' => 1200,
+        '0100' => 2400,
+        '0101' => 4800,
+        '0110' => 9600,
+        '0111' => 19200,
+        '1000' => 28800,
+        '1001' => 38400,
+        '1010' => 57600,
+        '1011' => 14400,
+    ];
+
+    const PRINT_MODES = [
+        '0' => "Rewind",
+        '1' => "Peel-Off",
+        '2' => "Tear-Off",
+        '3' => "Cutter",
+        '4' => "Applicator",
+        '5' => "Delayed cut",
+        '6' => "Linerless Peel",
+        '7' => "Linerless Rewind",
+        '8' => "Partial Cutter",
+        '9' => "RFID",
+        'K' => "Kiosk",
+        'S' => "Stream",
+    ];
+    
+    /**
+     * This is the parsed ordered list of values from the `~HS` command. Use the `POS_*` constants to determine the
+     * correct index of the value needed.
+     *
+     * e.g.
+     * ```
+     * Array
+     * (
+     *   [0] => 030
+     *   [1] => 0
+     *   [2] => 0
+     *   [3] => 0482
+     *   [4] => 000
+     *   [5] => 0
+     *   //...
+     *```
+     *
+     * @var array
+     */
+    private $data;
+
+    public function __construct($arrayData)
+    {
+        $this->data = $arrayData;
+    }
+
+    /**
+     * Creates a new object from the raw output of the `~HS` command.
+     *
+     * @param string $response
+     * @return PrinterStatus
+     */
+    public static function createFromRawResponse(string $response): self
+    {
+        // Strip out STX ETX characters
+        $response = preg_replace('/[\x02\x03]/','',$response);
+
+        // Split the lines
+        $lines = explode("\r\n", $response);
+
+        // Split the values of each line and merge them together
+        $data = [];
+        foreach ($lines as $line) {
+            $data = array_merge($data, explode(',', $line));
+        }
+
+        return new static($data);
+    }
+
+    /**
+     * Returns the current baud rate of the printer.
+     *
+     * @return int
+     */
+    public function getBaudRate() : int
+    {
+        $code =
+            $this->getCommBit(8) .
+            $this->getCommBit(2) .
+            $this->getCommBit(1) .
+            $this->getCommBit(0);
+
+        return self::BAUD_CODES[$code];
+    }
+
+    /**
+     * Returns the type of handshake the printer is configured for. ('Xon/Xoff' or 'DTR')
+     *
+     * @return string
+     */
+    public function getHandshakeType() : string
+    {
+        return !!$this->getCommBit(7)  ? 'Xon/Xoff' : 'DTR';
```

Comment:
I believe this is backwards since it will return "Xon/Xoff" for 1 but the docs say 1 should be DTR.

Seconding the mapping suggestion. 😉

Reply:
Fixed in e154f07c7a09db20067bd832c02b91f7c0532a12